source/deprecations/ember-data/v2.x.html.md

```diff
@@ -517,3 +517,95 @@ let author = this.store.peekRecord('user', 456);
 
 post.set('author', author);
 ```
+
+#### Arguments in Component Lifecycle Hooks
+
+##### until: 2.13.0
+##### id: ember-views.lifecycle-hook-arguments
+
+It is possible for component lifecycle hooks `didInitAttrs`, `didReceiveAttrs`, and `didUpdateAttrs` to receive arguments, but this functionality is part of private API, which is subject to change without notice. Those arguments will be removed because their use is harmful to component performance. However, since the arguments were accessible and some apps may be unknowingly depending on private API, this change will follow the formal deprecation process. Alternative approaches for all three hooks are below:
+
+##### `didInitAttrs` arguments
+
+Since this lifecycle hook is [already deprecated](http://emberjs.com/deprecations/v2.x/#toc_ember-component-didinitattrs), we suggest taking this chance to address two deprecations at the same time.
+
+Imagine you have a component that stores a timestamp when it is initialized.
+
+Before:
+
+``` javascript
+Ember.Component.extend({
+  didInitAttrs({ attrs }) {
+    this.set('initialTimestamp', attrs.timestamp);
+  }
+});
+```
+After:
+
+``` javascript
+Ember.Component.extend({
+  init() {
+    this._super(...arguments);
+
+    this.set('initialTimestamp', this.get('timestamp'));
+  }
+});
+```
+##### `didReceiveAttrs` arguments
+
+Let's say you want to animate a thermometer widget showing the change between today's high and low temperatures.
+
+Before:
+
+``` javascript
+Ember.Component.extend({
+  didReceiveAttrs({ highTemp, lowTemp }) {
```

Comment:
You can't change this, or the destructuring won't work. `didReceiveAttrs` receives an object with `oldAttrs` and `newAttrs` attributes, so:

```javascript
didReceiveAttrs({ oldAttrs, newAttrs }) {
}
```

Is equivalent to:

```javascript
didReceiveAttrs(attrObj) {
  let oldAttrs = attrObj.oldAttrs;
  let newAttrs = attrObj.newAttrs;
}
```

Did you mean to destructure oldAttrs/newAttrs themselves, like `didReceiveAttrs({ newAttrs: { lowTemp, highTemp }}) {`?